### PR TITLE
log epoch in metrics file and tensorboard

### PR DESCRIPTION
**Pull request by fhieber**
#### Pull Request Checklist ##
- [x] Changes are complete (if posting work-in-progress code, prefix your pull request title with '[WIP]'
until you can check this box.
- [x] Unit tests pass (`pytest`)
- [x] Were system tests modified? If so did you run these at least 5 times to account for the variation across runs?
- [x] System tests pass (`pytest test/system`)
- [x] Passed code style checking (`./pre-commit.sh` or manual run of pylint & mypy)
- [x] You have considered writing a test

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sockeye/utils.py [hunk 1]
```diff
@@ -684,7 +684,7 @@ def write_metrics_file(metrics: List[Dict[str, Any]], path: str):
     """
     with open(path, 'w') as metrics_out:
         for checkpoint, metric_dict in enumerate(metrics, 1):
-            metrics_str = "\t".join(["%s=%.6f" % (name, value) for name, value in sorted(metric_dict.items())])
+            metrics_str = "\t".join(["%s=%s" % (name, str(value)) for name, value in sorted(metric_dict.items())])
```
Comment (davvil) [inline sockeye/utils.py]: Why this conversion to string? Because epoch is an int? I think it would look cleaner with a format() call.

Comment (fhieber) [reply sockeye/utils.py]: with string conversion each type is printed more nicely. Ints without a digit and floats with as many digits as needed.

Comment (davvil) [reply sockeye/utils.py]: "{}={}".format(name, value) would do the same. It's nitpicking, but I prefer the syntax.

Comment (fhieber) [reply sockeye/utils.py]: I typically prefer the old-style format strings, I find the `{}`'s hard to read. Will change though.

**Pull request merged**